python/cudf/cudf/_lib/orc.pyx

```diff
@@ -62,9 +76,137 @@ from pyarrow.lib import NativeFile
 from cudf._lib.utils import _index_level_name, generate_pandas_metadata
 
 
-cpdef read_raw_orc_statistics(filepath_or_buffer):
+cdef _parse_column_type_statistics(column_statistics stats):
+    # Initialize stats to return and parse stats blob
+    column_stats = {}
```

Comment:
I was going to suggest using a `defaultdict` to avoid having to assign None manually to every value, but it looks like we eventually explicitly iterate over the contained values so that doesn't work. It would maybe be better to just not include all the None values at all in the final output? But that's an API breaking changes so we don't have to think about it for this PR.

Reply:
I am trying to match the previous logic exactly. Sometimes the values were set to `None` and sometimes they weren't set at all. If this could be normalized, then yes, a `defaultdict` would be fine, but I don't want to make behavior changes in this PR if possible.